sql/core/src/main/scala/org/apache/spark/sql/execution/joins/BroadcastHashJoinExec.scala

```diff
@@ -60,6 +60,26 @@ case class BroadcastHashJoinExec(
     }
   }
 
+  override def outputPartitioning: Partitioning = {
+    def buildKeys: Seq[Expression] = buildSide match {
+      case BuildLeft => leftKeys
+      case BuildRight => rightKeys
+    }
+
+    joinType match {
+      case _: InnerLike =>
```

Comment:
Just a question: Could we fix this issue in the `EnsureRequirements` side if this `outputPartitioning` get complicated?

Reply:
Sorry, I didn't understand the question. Could you clarify a bit?

Reply:
NVM, on second thought, its difficult to hanlde this issue in that side.